packages/admin/cms-admin/src/config/ApiConfigContext.tsx

```diff
@@ -0,0 +1,25 @@
+// TODO Remove React import once https://github.com/vivid-planet/comet/pull/2526 has been merged.
+import React, { createContext, PropsWithChildren, useContext } from "react";
+
+type ApiConfig = {
+    url: string;
+};
+
+const ApiConfigContext = createContext<ApiConfig | undefined>(undefined);
+
+function ApiConfigProvider({ config, children }: PropsWithChildren<{ config: ApiConfig }>) {
+    return <ApiConfigContext.Provider value={config}>{children}</ApiConfigContext.Provider>;
+}
+
+function useApiConfig() {
+    const context = useContext(ApiConfigContext);
+
+    if (!context) {
+        throw new Error("No ApiConfigContext instance can be found. Please ensure that you have called `ApiConfigProvider` higher up in your tree.");
+    }
+
+    return context;
+}
+
+export type { ApiConfig };
+export { ApiConfigProvider, useApiConfig };
```

Comment:
I don't really like having additional lines for exports, when you can just add export before the definition... do you prefer this for some reason? 

Don't care too much about this, just curious and prefer consistency 🙂

Reply:
> do you prefer this for some reason?

I like to do this because then all of a file's exports are in a single location, making it easier to understand the module boundaries. Imports at the top, exports at the bottom.

> prefer consistency

👍🏼 I can change it back once we decide which approach to use